src/imitation/data/rollout.py

```diff
@@ -419,13 +437,17 @@ def generate_trajectories(
     #
     # To start with, all environments are active.
     active = np.ones(venv.num_envs, dtype=bool)
-    assert isinstance(obs, np.ndarray), "Dict/tuple observations are not supported."
     state = None
     dones = np.zeros(venv.num_envs, dtype=bool)
     while np.any(active):
+        # policy gets unwrapped observations (eg as dict, not dictobs)
         acts, state = get_actions(obs, state, dones)
         obs, rews, dones, infos = venv.step(acts)
-        assert isinstance(obs, np.ndarray)
+        assert isinstance(
```

Comment:
Observation space shouldn't change type so might be overkill to assert this here as well as in line 417, although is harmless to do so and would catch a space-changing environment bug.

Reply:
agree. It's largely here to make the typechecker happy